src/screens/Home/components/Card/index.tsx

```diff
@@ -5,52 +5,64 @@ import {
 import { StackNavigationProp } from '@react-navigation/stack';
 import { Card } from '../../../../components';
 import * as S from './styles';
+import { formatMoney } from '../../../../utils/helper';
+import { useAuth } from '../../../../hooks/useAuth';
+import { handleBookmark } from '../../../../services/bookmark';
+import api from '../../../../services/api';
 
 interface DataCardProps {
   id: number
   title: string
   image: string
-  category: {
-    name: string
-  }
+  price: number
+  description: string
+  createdAt: string
 }
 interface TechnologyCardProps {
   data: DataCardProps
   style: StyleProp<any>
   loading: boolean
   navigation: StackNavigationProp<any, any>
 }
-
 interface FavoriteProps {
-  // eslint-disable-next-line react/require-default-props
   favorite?: boolean
+  technologyId: string
 }
 
-const Label = (): JSX.Element => (
-  <S.LabelWrapper>
-    <S.LabelText>Semiárido</S.LabelText>
-  </S.LabelWrapper>
-);
-
-const Favorite = ({ favorite }: FavoriteProps): JSX.Element => {
+const Favorite = ({ favorite, technologyId }: FavoriteProps): JSX.Element => {
   const [state, setState] = useState(favorite);
   const animatePulse = new Animated.Value(1);
   const scale = animatePulse.interpolate({
     inputRange: [0, 1],
     outputRange: [1.1, 1],
   });
 
+  const { user } = useAuth();
+
   useEffect(() => {
+    const isLiked = user?.bookmarks?.some((bookmark) => bookmark.id.toString() === technologyId);
+    setState(isLiked);
+
     Animated.timing(animatePulse, {
       toValue: state ? 0 : 1,
       duration: state ? 100 : 50,
       easing: Easing.linear,
       useNativeDriver: true,
     }).start();
-  }, [state]);
+  }, []);
+
+  const handleFavoriteClick = async () => {
+    setState((previousState) => !previousState);
+
+    await handleBookmark({
+      active: state,
+      technologyId,
+      userId: user.id,
+    });
+  };
```

Comment:
Here, we need to pay attention to the state update. The state can be updated only after the handleBookmark is executed, so the state value has been sent wrong to the API.

_setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall._

```suggestion
    const active = !state;
    setState(active);
    
    await handleBookmark({
      active,
      technologyId,
      userId: user.id,
    });
  };
```

Credits to @luizeboli, the React master 👏 

More about it in: https://reactjs.org/docs/react-component.html#setstate

Reply:
Yeah you're right, except for 'send the wrong value to the API', in this case it would not happen due setState inside useEffect .

You can also call setState after handleBookmark to achieve the same result and improve readability